src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs

```diff
@@ -67,20 +67,24 @@ protected bool FindResourceDesigner (AssemblyDefinition assembly, bool mainAppli
 			return false;
 		}
 
-		protected void ClearDesignerClass (TypeDefinition designer)
+		protected void ClearDesignerClass (TypeDefinition designer, bool completely = false)
```

Comment:
What are the semantics for `int[]` properties in `_Microsoft.Android.Resource.Designer.dll`?  This feels like something we should call out at least in the commit message.  Does it always create a new array instance:

```csharp
public static int[] myStyleable => new[] {1, 2, 3, 4};
```

or does it cache the array?

```csharp
private static readonly int[] myStylable_data = new[]{1, 2, 3, 4};
public static int[] myStyleable => myStylable_data;
```

Codegen-wise, I imagine the former is easier.  It's also *safer*, in case someone does `Resource.Stylable.myStylable[0] = 42`; the changes won't be persisted.  However, it also implies more memory allocations, as ever property invocation will allocate memory.

Documentation wise, we should say "don't modify this array"; it's unfortunate that we can't return an `IReadOnlyList<int>` for these properties, but that wouldn't work transparently. :-(

Reply:
Based on reading the Cecil-using codegen, it caches the array.  A new array is not always created.

This *should* be explicitly mentioned somewhere; "errant" modification of the array a'la `Resource.Stylable.myStylable[0] = 42` *will* be globally visible, and could screw things up.  If someone does that, it's on them, but we should at least mention "don't do that!" somewhere.

Reply:
Would that need to be in the property docs?

Reply:
Updated the commit message to mention Styleable array resources.